src/common.c

```diff
@@ -20,155 +20,166 @@
+        ++q;
+    return !*q && !(((p - q)) & 1);
 }
+
 int ishexlc(const char *q)
 {
-	const char *p=q;
-	while (atoi16l[ARCH_INDEX(*q)] != 0x7F)
-		++q;
-	return !*q && !(((p-q))&1);
+    const char *p = q;
+    while (atoi16l[ARCH_INDEX(*q)] != 0x7F)
+        ++q;
+    return !*q && !(((p - q)) & 1);
 }
 
 int ishexn(const char *q, int n)
 {
-	const char *p=q;
-	while (atoi16[ARCH_INDEX(*q)] != 0x7F)
-		++q;
-	return (q-p) >= n;
+    const char *p = q;
+    while (atoi16[ARCH_INDEX(*q)] != 0x7F)
+        ++q;
+    return (q - p) >= n;
 }
+
 int ishexucn(const char *q, int n)
 {
-	const char *p=q;
-	while (atoi16u[ARCH_INDEX(*q)] != 0x7F)
-		++q;
-	return (q-p) >= n;
+    const char *p = q;
+    while (atoi16u[ARCH_INDEX(*q)] != 0x7F)
+        ++q;
+    return (q - p) >= n;
 }
+
 int ishexlcn(const char *q, int n)
 {
-	const char *p=q;
-	while (atoi16l[ARCH_INDEX(*q)] != 0x7F)
-		++q;
-	return (q-p) >= n;
+    const char *p = q;
+    while (atoi16l[ARCH_INDEX(*q)] != 0x7F)
+        ++q;
+    return (q - p) >= n;
 }
 
-int ishexuc_oddOK(const char *q) {
-	while (atoi16[ARCH_INDEX(*q)] != 0x7F) {
-		if (*q >= 'a' && *q <= 'f') return 0;
-		++q;
-	}
-	return !*q ;
+int ishexuc_oddOK(const char *q)
+{
+    while (atoi16[ARCH_INDEX(*q)] != 0x7F)
+    {
+        if (*q >= 'a' && *q <= 'f') return 0;
+        ++q;
+    }
+    return !*q;
 }
-int ishexlc_oddOK(const char *q) {
-	while (atoi16[ARCH_INDEX(*q)] != 0x7F) {
-		if (*q >= 'A' && *q <= 'F') return 0;
-		++q;
-	}
-	return !*q ;
+
+int ishexlc_oddOK(const char *q)
+{
+    while (atoi16[ARCH_INDEX(*q)] != 0x7F)
+    {
+        if (*q >= 'A' && *q <= 'F') return 0;
+        ++q;
+    }
+    return !*q;
 }
 
 static MAYBE_INLINE size_t _hexlen(const char *q, unsigned char dic[0x100], int *extra_chars)
 {
-	const char *s = q;
-	size_t len = strlen(q);
-	if (len&1) --len;
+    const char *s = q;
+    size_t len = strlen(q);
+    if (len & 1) --len;
 
-	while (dic[ARCH_INDEX(*q)] != 0x7F)
-		++q;
-	if ((size_t)(q - s)&1) --q;
-	if (extra_chars)
-		*extra_chars = (*q != 0);
-	return (q - s);
+    while (dic[ARCH_INDEX(*q)] != 0x7F)
+        ++q;
+    if ((size_t) (q - s) & 1) --q;
+    if (extra_chars)
+        *extra_chars = (*q != 0);
+    return (q - s);
 }
+
 size_t hexlen(const char *q, int *extra_chars)
 {
-	return _hexlen(q, atoi16, extra_chars);
+    return _hexlen(q, atoi16, extra_chars);
 }
+
 size_t hexlenu(const char *q, int *extra_chars)
 {
-	return _hexlen(q, atoi16u, extra_chars);
+    return _hexlen(q, atoi16u, extra_chars);
 }
+
 size_t hexlenl(const char *q, int *extra_chars)
 {
-	return _hexlen(q, atoi16l, extra_chars);
+    return _hexlen(q, atoi16l, extra_chars);
 }
 
-int isdec(const char *q)
-{
-	char buf[24];
-	int x = atoi(q);
-	sprintf(buf, "%d", x);
-	return !strcmp(q,buf) && *q != '-';
-}
+
 int isdec_negok(const char *q)
 {
-	char buf[24];
-	int x = atoi(q);
-	sprintf(buf, "%d", x);
-	return !strcmp(q,buf);
+    if (q && *q == '-')
```

Comment:
Why did you add the `q &&` here? If someone calls one of these functions with `NULL`, we should crash, not hide the caller's error.

Reply:
@solardiz oh ok. I thought we should fail on `NULL`. I will fix that.